metaflow/plugins/exit_hook/exit_hook_decorator.py

```diff
@@ -0,0 +1,32 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+
+
+class ExitHookDecorator(FlowDecorator):
+    name = "exit_hook"
+    allow_multiple = True
+
+    defaults = {
+        "image": None,
+        "on_success": [],
+        "on_error": [],
+    }
+
+    def flow_init(
+        self, flow, graph, environment, flow_datastore, metadata, logger, echo, options
+    ):
+        on_success = self.attributes["on_success"]
+        on_error = self.attributes["on_error"]
+
+        if not on_success and not on_error:
+            raise MetaflowException(
+                "Choose at least one of the options on_success/on_error"
+            )
+
+        self.success_hooks = []
+        self.error_hooks = []
+        for success_fn in self.attributes["on_success"]:
+            self.success_hooks.append(success_fn.__name__)
```

Comment:
it would be nice to be able to take strings too. I think that would have the benefit of being more easily configurable (config could contain the string name for the hook)

For functions, I would actually store the full path because that would make it compatible (or more compatible) with the new packaging AND you wouldn't have to load your flow file to run the hooks (flow files can contain other stuff that you may not need/want/can use given your mention that the image may be different, etc.). To execute we could then load *just* what is needed.